tests/integration/test_tf_symbolic.py

```diff
@@ -565,7 +565,10 @@ def test_eval_true_state_all_fock_probs(self, setup_eng, cutoff, batch_size, tol
             Dgate(-ALPHA) | q[1]
 
         state = eng.run(prog).state
-        probs = state.all_fock_probs(eval=True).flatten()
```

Comment:
This entire file needs a closer inspection. Most of the tests here were testing the `eval=True/False`. They are still useful as integration tests, and as tests that tensorflow returns tf objects, but the scope of the file has changed significantly now that we don't even support the `eval` keyword behaviour for TF2.0

Reply:
Also, the `mark.xfail`ed test can be dropped (we no longer support `sessions` or `feed_dict`s). The "reason" message is also a bit misleading

Reply:
@co9olguy: have re-written this tests, making it more of an integration test, as well as removing the redundant tests. I have also added some gradient integration tests, to make sure everything works as it should.